test/e2e_env/kubernetes/gateway/delegated.go

```diff
@@ -0,0 +1,77 @@
+package gateway
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	. "github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/framework/deployments/democlient"
+	"github.com/kumahq/kuma/test/framework/deployments/kic"
+	"github.com/kumahq/kuma/test/framework/deployments/testserver"
+	"github.com/kumahq/kuma/test/framework/envs/kubernetes"
+)
+
+func Delegated() {
+	namespace := "delegated-gateway"
+	namespaceOutsideMesh := "delegated-gateway-outside-mesh"
+	mesh := "delegated-gateway"
+
+	var kicIP string
+
+	BeforeAll(func() {
+		err := NewClusterSetup().
+			Install(MTLSMeshKubernetes(mesh)).
+			Install(NamespaceWithSidecarInjection(namespace)).
+			Install(Namespace(namespaceOutsideMesh)).
+			Install(democlient.Install(
+				democlient.WithNamespace(namespaceOutsideMesh),
+			)).
+			Install(testserver.Install(
+				testserver.WithMesh(mesh),
+				testserver.WithNamespace(namespace),
+				testserver.WithName("test-server"),
+			)).
+			Install(kic.KongIngressController(
```

Comment:
To verify if it works with arm64. I am not sure if we updated the version

Reply:
It doesn't work with arm64, so I updated the labels